components/relevancy/src/populate_interests.rs

```diff
@@ -2,69 +2,119 @@
+                return Err(Error::FetchInterestDataError);
             }
-            dao.err_if_interrupted()?;
-        }
+        };
         Ok(())
     })
 }
 
 /// Fetch the interest data
-fn fetch_interest_data() -> std::io::Result<Vec<(UrlHash, Interest)>> {
-    // TODO: this hack should be replaced with something that fetches from remote settings.
-    // It should ideally check for interruption while fetching the data.
-    let bytes = include_bytes!("../test-data");
-    let mut reader = Cursor::new(&bytes);
+fn fetch_interest_data() -> Result<Vec<(Interest, UrlHash)>> {
+    let rs = Client::new(RemoteSettingsConfig {
+        collection_name: REMOTE_SETTINGS_COLLECTION.to_string(),
+        server: Some(RemoteSettingsServer::Prod),
+        server_url: None,
+        bucket_name: None,
+    })?;
+    let remote_settings_response = rs.get_records()?;
     let mut data = vec![];
+    for record in remote_settings_response.records {
+        let record_fields: RelevancyRecord =
+            serde_json::from_value(serde_json::Value::Object(record.fields.clone()))?;
+
+        let attachment_data = match &record.attachment {
+            None => return Err(Error::FetchInterestDataError),
+            Some(a) => rs.get_attachment(&a.location)?,
+        };
 
-    // Loop over all possible interests
-    for interest in Interest::all() {
-        // read the count
-        let mut buf = [0u8; 4];
-        reader.read_exact(&mut buf)?;
-        let count = u32::from_le_bytes(buf);
-        for _ in 0..count {
-            let mut url_hash: UrlHash = [0u8; 16];
-            reader.read_exact(&mut url_hash)?;
-            data.push((url_hash, interest));
-        }
+        let parsed_attachment_data =
+            serde_json::from_slice::<Vec<RelevancyAttachmentData>>(&attachment_data)?;
+        parse_record(record_fields, parsed_attachment_data, &mut data)?;
     }
     Ok(data)
 }
 
+/// Parse record and extract interest info
+fn parse_record(
+    record_fields: RelevancyRecord,
+    parsed_attachment_data: Vec<RelevancyAttachmentData>,
+    data: &mut Vec<(Interest, UrlHash)>,
+) -> Result<()> {
+    let custom_details = record_fields.record_custom_details;
+    let category_code = custom_details.category_to_domains.category_code;
+
+    for attachment_data in parsed_attachment_data {
+        let hash_url = STANDARD
+            .decode(attachment_data.domain)
+            .map_err(|_| Error::Base64DecodeError("Invalid base64 error".to_string()))?;
+        let url_hash = hash_url.try_into().map_err(|_| {
+            Error::Base64DecodeError("Base64 string has wrong number of bytes".to_string())
+        })?;
+        let interest = Interest::try_from(category_code as u32)?;
+        data.push((interest, url_hash));
+    }
+    Ok(())
+}
+
+/// Insert Interests into Db
+fn insert_interest_data(data: Vec<(Interest, UrlHash)>, dao: &mut RelevancyDao) {
+    for (interest, hash_url) in data {
```

Comment:
Could you add the interruption support back here?  That means processing the data in chunks, and calling `dao.err_if_interrupted()?` so that we error out early if the operation is interrupted.

Reply:
i realized i totally read this wrong yesterday